cmd/trace-agent/config/config.go

```diff
@@ -290,7 +291,10 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 			c.Obfuscation.CreditCards.Luhn = coreconfig.Datadog.GetBool("apm_config.obfuscation.credit_cards.luhn")
 		}
 	}
-
+	c.Obfuscation.AppSec = config.AppSecObfuscation{
+		ParameterKeyRegexp:   obfuscate.CompileRegexp(coreconfig.Datadog.GetString("appsec_config.obfuscation.parameter_key_regexp")),
```

Comment:
The `obfuscate` module is stand alone. I don't think it requires new API for compiling regular expressions. Why not just pass a string here and let the constructor process it as previously suggested?

Reply:
What other module that is shared between `pkg/trace/config` and `cmd/trace-agent/config` do you suggest me?

Reply:
Why not just leave it here in the same file as a private method?

Reply:
Done in https://github.com/DataDog/datadog-agent/pull/11324/commits/8f7ce7b12e26c8ed0c9f12f70f0210f7f57d891f